### PR TITLE
2394: Adding timelog delete button

### PR DESCRIPTION
**Pull request by jeppekroghitk**
#### Link to ticket

[2394](https://leantime.itkdev.dk/#/tickets/showTicket/2394)

#### Description

Added delete button to time log modal
Added possibility to change date when editing time log
Added title describing the exclamation mark on logs with no description

#### Screenshot of the result

N/A

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

If you have any further comments or questions for the reviewer please add them here.

### CODE REVIEW

File: Templates/timetable.blade.php [hunk 1]
```diff
@@ -120,7 +129,7 @@
 
             {{-- copy paste from https://www.w3schools.com/howto/howto_js_filter_dropdown.asp - also entries in timeTable.css and timeTable.js --}}
             <div class="timetable-ticket-search">
-                <input class="timetable-ticket-input" type="text" data-placeholder="Search tickets.." data-loading="Filtering tickets.." placeholder="Search todo.." />
+                <input class="timetable-ticket-input" type="text" data-placeholder="Search tickets.." data-loading="Filtering already logged tickets.." placeholder="Search todo.." />
```
Comment (sinejespersen) [inline Templates/timetable.blade.php]: strings in language file?

File: assets/timeTable.js [hunk 2]
```diff
@@ -266,17 +276,17 @@ jQuery(document).ready(function ($) {
...
-      const { text, children } = obj;
+      const { id, text, children } = obj;
 
       const lowerCaseQuery = query.toLowerCase();
 
       let results = [];
 
-      // Checks if `obj`'s `text` contains `lowerCaseQuery` and not already added to timetable.
       if (
         "text" in obj &&
```
Comment (sinejespersen) [inline assets/timeTable.js:286]: Why remove above comment?

File: assets/timeTable.js [hunk 3]
```diff
@@ -334,12 +344,18 @@ jQuery(document).ready(function ($) {
...
+        this.modalDeleteButton.hide();
+      }
+
       this.modalInputTimesheetId.val(id);
       this.modalInputTicketId.val(ticket.id);
       this.modalInputTicketName.val(ticket.text).attr("disabled", "disabled");
       this.modalInputHours.val(hours);
       this.modalTextareaDescription.val(description);
-      this.modalInputDate.val(date).attr("disabled", "disabled");
+      this.modalInputDate.val(date); //.attr("disabled", "disabled");
```
Comment (sinejespersen) [inline assets/timeTable.js]: delete?

File: assets/timeTable.js [hunk 4]
```diff
@@ -448,6 +464,30 @@ jQuery(document).ready(function ($) {
         .html(ticketsLastUpdatedElement);
     }
 
+    deleteTimeEntry() {
+      const timesheetId = this.modalInputTimesheetId.val();
+      $(this.modalDeleteButton).text($(this.modalDeleteButton).data("loading"));
+      fetch("", {
```
Comment (sinejespersen) [inline assets/timeTable.js]: Empty fetch?

File: Templates/timetable.blade.php [hunk 5]
```diff
@@ -75,10 +75,19 @@
...
+                                        data-hours="{{ $hours }}"
+                                        data-description="{{ $description }}"
+                                        data-date="{{$weekDate->format('Y-m-d')}}"
+                                        title="{{ isset($hours) && trim($description) === '' ? __("timeTable.description_missing") : '' }}"
+                                    >
                                         <span>{{ $hours }}</span>
-                                        @if (isset($hours) && $description === '')
-                                            <span class="fa fa-circle-exclamation"></span>
+                                        @if (isset($hours) && trim($description) === '')
+                                            <span  class="fa fa-circle-exclamation"></span>
```
Comment (sinejespersen) [inline Templates/timetable.blade.php]:
```suggestion
                                            <span class="fa fa-circle-exclamation"></span>
```

File: Repositories/TimeTable.php [hunk 6]
```diff
@@ -35,7 +35,7 @@ public function getUniqueTicketIds(CarbonImmutable $dateFrom, CarbonImmutable $d
         $sql = 'SELECT DISTINCT
         timesheet.ticketId
         FROM zp_timesheets AS timesheet
```
Comment (sinejespersen) [inline Repositories/TimeTable.php:37]: Why not between?

**Review by sinejespersen — CHANGES_REQUESTED** [review]

File: assets/timeTable.js (hunk 2 above)
Comment (jeppekroghitk) [reply assets/timeTable.js:286]: Restored

**Pull request merged**